app/controllers/observer_controller/indexes.rb

```diff
@@ -218,12 +218,11 @@ def map_observations
 
     unless locations.empty?
       # Eager-load corresponding locations.
-      @locations = Location.connection.select_rows(%(
-        SELECT id, name, north, south, east, west FROM locations
-        WHERE id IN (#{locations.keys.sort.map(&:to_s).join(",")})
-      )).map do |id, *the_rest|
-        locations[id.to_i] = MinimalMapLocation.new(id, *the_rest)
-      end
+      @locations = Location.where(id: locations.keys.sort).
```

Comment:
I think the catalyst for changing this was a brakeman complaint about possible XSS.